pkg/controller/datavolume/controller-base.go

```diff
@@ -323,18 +346,123 @@ func (r ReconcilerBase) syncUpdate(log logr.Logger, syncRes *dataVolumeSyncResul
 	return nil
 }
 
-func (r ReconcilerBase) handlePrePopulation(dv *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) {
+func (r *ReconcilerBase) handleStaticVolume(syncRes *dataVolumeSyncResult, log logr.Logger) error {
+	if _, ok := syncRes.dvMutated.Annotations[cc.AnnCheckStaticVolume]; !ok {
+		return nil
+	}
+
+	if syncRes.pvc == nil {
+		volumes, err := r.getAvailableVolumesForDV(syncRes, log)
+		if err != nil {
+			return err
+		}
+
+		if len(volumes) == 0 {
+			log.Info("No PVs for DV")
+			return nil
+		}
+
+		if err := r.handlePvcCreation(log, syncRes, func(_ *cdiv1.DataVolume, pvc *corev1.PersistentVolumeClaim) error {
+			bs, err := json.Marshal(volumes)
+			if err != nil {
+				return err
+			}
+			cc.AddAnnotation(pvc, cc.AnnPersistentVolumeList, string(bs))
+			return nil
+		}); err != nil {
+			return err
+		}
+
+		// set result to make sure callers don't do anything else in sync
+		syncRes.result = &reconcile.Result{}
+		return nil
+	}
+
+	volumeAnno, ok := syncRes.pvc.Annotations[cc.AnnPersistentVolumeList]
+	if !ok {
+		// etiher did not create the PVC here OR bind to expected PV succeeded
+		return nil
+	}
+
+	if syncRes.pvc.Spec.VolumeName == "" {
+		// set result to make sure callers don't do anything else in sync
+		syncRes.result = &reconcile.Result{}
+		return nil
+	}
+
+	var volumes []string
+	if err := json.Unmarshal([]byte(volumeAnno), &volumes); err != nil {
+		return err
+	}
+
+	for _, v := range volumes {
+		if v == syncRes.pvc.Spec.VolumeName {
+			pvcCpy := syncRes.pvc.DeepCopy()
```

Comment:
why use a copy if we don't need `DeepEqual` check here?

Reply:
Best practice to always mutate a copy especially in this case where the pvc reference lives beyond the scope of the call.  Imagine the update call fails, that annotation will stay in the shared reference even though that is not how it looks in etcd